libs/ledger-live-common/src/hooks/useAccountsWithFundsListener.ts

```diff
@@ -0,0 +1,71 @@
+import { useEffect, useMemo, useRef } from "react";
+import debounce from "lodash/debounce";
+import type { Account } from "@ledgerhq/types-live";
+
+/**
+ * This function returns true if the acccounts with funds changed between the accounts and the oldAccounts properties.
+ * Meaning that at least one account with a positive balance has been added, removed, the balance of an existing empty
+ * account is now positive or the balance of an existing account was positive and is now 0.
+ * Else this function returns false.
+ */
+export function hasAccountsWithFundsChanged(accounts: Account[], oldAccounts: Account[]): boolean {
+  for (const account of accounts) {
+    const matchingOldAccount = oldAccounts.find(acc => acc.id === account.id);
+
+    // Means that account has been added
+    if (!matchingOldAccount) {
+      if (account.balance.isGreaterThan(0)) {
+        return true;
+      } else {
+        continue;
+      }
+    }
+
+    // Means that account has been modified
+    const hasAccountBeenEmptied =
+      account?.balance.isZero() && matchingOldAccount?.balance.isGreaterThan(0);
+    const hasAccountReceivedFunds =
+      account?.balance.isGreaterThan(0) && matchingOldAccount?.balance.isZero();
+
+    if (hasAccountBeenEmptied || hasAccountReceivedFunds) {
+      return true;
+    }
```

Comment:
```suggestion
    return (hasAccountBeenEmptied || hasAccountReceivedFunds)
```
Could works :p

Reply:
If none of the two is true we need to execute the second for loop ;)